### PR TITLE
Bring changes from master to preview

### PR DESCRIPTION
**Pull request by abhipsaMisra**
SDK code changes:
- pnp support over Amqp
- support for DTDL component name to be passed via message system properties.

### CODE REVIEW

File: e2e/test/E2ETestBase.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            Stopwatch.Stop();
+            double elapsed = Stopwatch.Elapsed.TotalSeconds;
+            var properties = new Dictionary<string, string>
+            {
+                { LoggingPropertyNames.SecondsElapsed, elapsed.ToString() }
+            };
+            Logger.WriteLine($"Finished test - {TestContext.TestName}", SeverityLevel.Information, properties);
+            // As this is not an application that keeps running, explicitly flushing is required to ensure we do not lose any logs.
+            // The recommendation from AI is to wait for 5 seconds after flushing.
```
Comment (azabbasi) [inline e2e/test/E2ETestBase.cs:40]: in that case, isn't it better if the Logger.Flush method does the waiting?

Comment (azabbasi) [reply e2e/test/E2ETestBase.cs:40]: so we don't have to always remember to do Thread.Wait every time we Flush

Comment (azabbasi) [reply e2e/test/E2ETestBase.cs:40]: and this is a test logger ... so we should be fine.

Comment (abhipsaMisra) [reply e2e/test/E2ETestBase.cs:40]: good idea, @vinagesh Can we add the sleep in here directly: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/common/test/TestLogging.cs#L115

**Pull request merged**